numpy/ma/core.py

```diff
@@ -6309,6 +6309,11 @@ def copy(self, *args, **kwargs):
         # precedent for this with `np.bool_` scalars.
         return self
 
+    def __setattr__(self, attr, value):
+        if self is self.__singleton:
+            raise MaskError("Cannot change attributes of the masked singleton")
+        super(MaskedConstant, self).__setattr__(attr, value)
```

Comment:
Is this functionality needed? Maybe a note.

Reply:
It stops it from breaking subclasses, if anyone else is brave enough to do that (@mhvk?)

Reply:
No, while we do subclass `MaskedArray`, even we don't subclass this (as far as I know).

Reply:
Update: yes, this is needed in order to reassign `__class__` of duplicate instances in `__array_finalize__`